app/Controllers/Api/V1/CategoriesController.php

```
@@ -19,61 +21,156 @@
     public function update($id = null): ResponseInterface
     {
-        // TODO: validate, model update, return transformer result
-        // return $this->respond($this->transformer->transform($category));
-        return $this->fail('Not Implemented', 501);
+        $category = $this->model->find($id);
+
+        if (!$category) {
+            return $this->failNotFound("Category not found: $id");
+        }
+
+        if ((int)$category->tenant_id !== (int)auth()->user()->tenant_id) {
```

PHPStan (2.1.46)

[error] Access to an undefined property CodeIgniter\Shield\Entities\User::$tenant_id.
Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property

(property.notFound)